app/services/moves/updater.rb

```diff
@@ -44,20 +64,22 @@ def document_ids
     # 3. Frontend specifies empty person: update person to be nil
     # 4. Frontend does not include person relationship: don't update person at all
     def attributes
-      attributes = move_params.fetch(:attributes, {})
+      @attributes ||= begin
+        attributes = move_params.fetch(:attributes, {})
 
-      # TODO: to be removed once move profile migration complete
-      if person_attributes.present? && profile_attributes.nil?
-        person = Person.find_by(id: person_attributes.dig(:data, :id))
-        attributes[:profile] = person&.latest_profile
-      end
+        # TODO: to be removed once move profile migration complete
+        if person_attributes.present? && profile_attributes.nil?
+          person = Person.find_by(id: person_attributes.dig(:data, :id))
+          attributes[:profile] = person&.latest_profile
+        end
 
-      attributes[:profile] = Profile.find_by(id: profile_attributes.dig(:data, :id)) if profile_attributes.present?
+        attributes[:profile] = Profile.find_by(id: profile_attributes.dig(:data, :id)) if profile_attributes.present?
 
-      profile = attributes[:profile] || move.profile
-      profile.documents = Document.where(id: document_ids) unless document_attributes.nil?
+        profile = attributes[:profile] || move.profile
+        profile.documents = Document.where(id: document_ids) unless document_attributes.nil?
 
-      attributes
+        attributes
+      end
```

Comment:
Just to refresh my Ruby memory, we're doing this because we're manipulating the attributes in the `call` method?

Reply:
I'm memoizing the attributes here for efficiency - to avoid running the same code multiple times unnecessarily. The `begin` and `end` block store the result in the `@attributes` variable